packages/core/src/WorkflowExecute.ts

```diff
@@ -948,7 +948,7 @@ export class WorkflowExecute {
 			const returnPromise = (async () => {
 				try {
 					if (!this.additionalData.restartExecutionId) {
-						await this.executeHook('workflowExecuteBefore', [workflow]);
+						await this.executeHook('workflowExecuteBefore', [workflow, this.runExecutionData]);
```

Comment:
I'm wondering if we need to update the two other invocations of this hook as well, to avoid any weird edge-cases, like when credential permissions are revoked, and `permissionChecker.check` fails.

Reply:
let's fix this in another PR, to reduce the scope of this PR.